src/openshift/component.ts

```diff
@@ -362,9 +362,11 @@ export class Component extends OpenShiftItem {
                         Component.stateChanged.fire(component.contextPath);
                     },
                     onText(text: string) {
-                        if (cs.devStatus === ComponentContextState.DEV_STARTING && text.includes('[p]')) {
+                        if (cs.devStatus === ComponentContextState.DEV_STARTING) {
```

Comment:
This is bad. The odo component isn't running until the '[p]' shows up. With this current change, it lists the component as running before the source code is synced into the running pod. After odo syncs the source code, it still needs to perform whatever build steps the Devfile specifies, and set up port forwarding so that the user can actually access the component.

Also, we should change `devStatus` and fire the event indicating that `devStatus` changed at the same time. With this change, `devStatus` is changed to running as soon as odo outputs any text, and the event that indicates it changed is fired when "Pods is Running" is output.